File: Overlord.Core/Overlord.Core/Models/PlatoonModels.cs
using System.Text.Json.Serialization;

namespace Overlord.Core.Models;

/// <summary>
/// Equipment levels for platoons (body armor).
/// </summary>
[JsonConverter(typeof(JsonStringEnumConverter))]
public enum EquipmentLevel
{
    Civilian,   // 20,000 Credits, 0.5x modifier (no armor)
    Basic,      // 35,000 Credits, 1.0x modifier (light armor)
    Standard,   // 55,000 Credits, 1.5x modifier (medium armor)
    Advanced,   // 80,000 Credits, 2.0x modifier (heavy armor)
    Elite       // 109,000 Credits, 2.5x modifier (power armor)
}

/// <summary>
/// Weapon levels for platoons.
/// </summary>
[JsonConverter(typeof(JsonStringEnumConverter))]
public enum WeaponLevel
{
    Pistol,        // 5,000 Credits, 0.8x modifier
    Rifle,         // 10,000 Credits, 1.0x modifier
    AssaultRifle,  // 18,000 Credits, 1.3x modifier
    Plasma         // 30,000 Credits, 1.6x modifier
}

/// <summary>
/// Platoon costs based on equipment and weapon levels.
/// </summary>
public class PlatoonCosts
{
    /// <summary>
    /// Equipment cost by level.
    /// </summary>
    public static int GetEquipmentCost(EquipmentLevel level)
    {
        return level switch
        {
            EquipmentLevel.Civilian => 20000,
            EquipmentLevel.Basic => 35000,
            EquipmentLevel.Standard => 55000,
            EquipmentLevel.Advanced => 80000,
            EquipmentLevel.Elite => 109000,
            _ => 0
        };
    }

    /// <summary>
    /// Weapon cost by level.
    /// </summary>
    public static int GetWeaponCost(WeaponLevel level)
    {
        return level switch
        {
            WeaponLevel.Pistol => 5000,
            WeaponLevel.Rifle => 10000,
            WeaponLevel.AssaultRifle => 18000,
            WeaponLevel.Plasma => 30000,
            _ => 0
        };
    }

    /// <summary>
    /// Total cost for a platoon (equipment + weapon, one-time cost).
    /// </summary>
    public static int GetTotalCost(EquipmentLevel equipment, WeaponLevel weapon)
    {
        return GetEquipmentCost(equipment) + GetWeaponCost(weapon);
    }
}

/// <summary>
/// Combat modifiers for equipment and weapons.
/// </summary>
public class PlatoonModifiers
{
    /// <summary>
    /// Equipment combat modifier.
    /// </summary>
    public static float GetEquipmentModifier(EquipmentLevel level)
    {
        return level switch
        {
            EquipmentLevel.Civilian => 0.5f,
            EquipmentLevel.Basic => 1.0f,
            EquipmentLevel.Standard => 1.5f,
            EquipmentLevel.Advanced => 2.0f,
            EquipmentLevel.Elite => 2.5f,
            _ => 1.0f
        };
    }

    /// <summary>
    /// Weapon combat modifier.
    /// </summary>
    public static float GetWeaponModifier(WeaponLevel level)
    {
        return level switch
        {
            WeaponLevel.Pistol => 0.8f,
            WeaponLevel.Rifle => 1.0f,
            WeaponLevel.AssaultRifle => 1.3f,
            WeaponLevel.Plasma => 1.6f,
            _ => 1.0f
        };
    }

    /// <summary>
    /// Training modifier (0.0 to 1.0 based on training level).
    /// </summary>
    public static float GetTrainingModifier(int trainingLevel)
    {
        return trainingLevel / 100.0f;
    }

    /// <summary>
    /// Calculates total military strength for a platoon.
    /// Formula: Troops × Equipment × Weapon × Training
    /// </summary>
    public static int CalculateMilitaryStrength(int troops, EquipmentLevel equipment, WeaponLevel weapon, int trainingLevel)
    {
        float equipMod = GetEquipmentModifier(equipment);
        float weaponMod = GetWeaponModifier(weapon);
        float trainingMod = GetTrainingModifier(trainingLevel);

        return (int)Math.Floor(troops * equipMod * weaponMod * trainingMod);
    }
}

File: Overlord.Phaser/src/core/models/PlatoonModels.ts
import { EquipmentLevel, WeaponLevel } from './Enums';

/**
 * Platoon costs based on equipment and weapon levels.
 * REBALANCED: Costs reduced 5× to align with increased taxation income
 * Progression: Civilian (~3k) → Basic (~8k) → Standard (~13k) → Advanced (~20k) → Elite (~30k)
 */
export class PlatoonCosts {
  /**
   * Equipment cost by level.
   */
  public static getEquipmentCost(level: EquipmentLevel): number {
    switch (level) {
      case EquipmentLevel.Civilian:
        return 2500;   // Was 20,000 (÷8 for early game viability)
      case EquipmentLevel.Basic:
        return 6000;   // Was 35,000 (÷5.8)
      case EquipmentLevel.Standard:
        return 10000;  // Was 55,000 (÷5.5)
      case EquipmentLevel.Advanced:
        return 16000;  // Was 80,000 (÷5)
      case EquipmentLevel.Elite:
        return 25000;  // Was 109,000 (÷4.4)
      default:
        return 0;
    }
  }

  /**
   * Weapon cost by level.
   */
  public static getWeaponCost(level: WeaponLevel): number {
    switch (level) {
      case WeaponLevel.Pistol:
        return 500;    // Was 5,000 (÷10)
      case WeaponLevel.Rifle:
        return 2000;   // Was 10,000 (÷5)
      case WeaponLevel.AssaultRifle:
        return 3500;   // Was 18,000 (÷5.1)
      case WeaponLevel.Plasma:
        return 6000;   // Was 30,000 (÷5)
      default:
        return 0;
    }
  }

  /**
   * Total cost for a platoon (equipment + weapon, one-time cost).
   */
  public static getTotalCost(equipment: EquipmentLevel, weapon: WeaponLevel): number {
    return PlatoonCosts.getEquipmentCost(equipment) + PlatoonCosts.getWeaponCost(weapon);
  }
}

/**
 * Combat modifiers for equipment and weapons.
 */
export class PlatoonModifiers {
  /**
   * Equipment combat modifier.
   */
  public static getEquipmentModifier(level: EquipmentLevel): number {
    switch (level) {
      case EquipmentLevel.Civilian:
        return 0.5;
      case EquipmentLevel.Basic:
        return 1.0;
      case EquipmentLevel.Standard:
        return 1.5;
      case EquipmentLevel.Advanced:
        return 2.0;
      case EquipmentLevel.Elite:
        return 2.5;
      default:
        return 1.0;
    }
  }

  /**
   * Weapon combat modifier.
   */
  public static getWeaponModifier(level: WeaponLevel): number {
    switch (level) {
      case WeaponLevel.Pistol:
        return 0.8;
      case WeaponLevel.Rifle:
        return 1.0;
      case WeaponLevel.AssaultRifle:
        return 1.3;
      case WeaponLevel.Plasma:
        return 1.6;
      default:
        return 1.0;
    }
  }

  /**
   * Training modifier (0.0 to 1.0 based on training level).
   */
  public static getTrainingModifier(trainingLevel: number): number {
    return trainingLevel / 100.0;
  }

  /**
   * Calculates total military strength for a platoon.
   * Formula: Troops × Equipment × Weapon × Training
   */
  public static calculateMilitaryStrength(
    troops: number,
    equipment: EquipmentLevel,
    weapon: WeaponLevel,
    trainingLevel: number,
  ): number {
    const equipMod = PlatoonModifiers.getEquipmentModifier(equipment);
    const weaponMod = PlatoonModifiers.getWeaponModifier(weapon);
    const trainingMod = PlatoonModifiers.getTrainingModifier(trainingLevel);

    return Math.floor(troops * equipMod * weaponMod * trainingMod);
  }
}
